apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -3104,6 +3114,50 @@ post_tx_parse_id(verify_id_match, {MaybeTXID, Req, TX}) ->
 					end
 			end
 	end.
+should_import_missing_data(TXID, Req, Pid, Encoding) ->
+    case read_complete_body(Req, Pid) of
+        {ok, Body, _Req2} ->
+            case parse_tx_from_body(Body, Encoding) of
+                {ok, TX} when TX#tx.id == TXID ->
+                    case tx_has_missing_data(TX) of
+                        true ->
+                            {import_data, TX};
+                        false ->
+                            no_import
+                    end;
+                _ ->
+                    no_import
+            end;
+        _ ->
+            no_import
+    end.
+
+parse_tx_from_body(Body, json) ->
+    case catch ar_serialize:json_struct_to_tx(Body) of
+        {'EXIT', _} -> error;
+        {error, _} -> error;
+        TX -> {ok, TX}
+    end;
+parse_tx_from_body(Body, binary) ->
+    case catch ar_serialize:binary_to_tx(Body) of
+        {'EXIT', _} -> error;
+        {error, _} -> error;
+        {ok, TX} -> {ok, TX}
+    end.
+
+tx_has_missing_data(#tx{ format = 2, data_size = DataSize, data = Data, id = ID })
+        when DataSize > 0, byte_size(Data) > 0 ->
+    % Check if we have the data for this tx
+    case ar_storage:read_tx_data(ID) of
```

Comment:
We do not use the file storage anymore, use `ar_data_sync:get_tx_data/1` to fetch the transaction data.

In any case, we are interested in the mempool here, we do not put unconfirmed transactions on disk. Also, we should not do any IO in the POST /tx handler for performance reasons.

I think, the ignore registry upgrade I propose above naturally solves this problem.

Reply:
this function is now gone in favour of ignore registry